resolver/addressable_resolver.go

```diff
@@ -184,30 +247,24 @@ func (r *URIResolver) URIFromObjectReference(ctx context.Context, ref *corev1.Ob
 		return nil, fmt.Errorf("failed to get object %s/%s: %w", ref.Namespace, ref.Name, err)
 	}
 
-	// K8s Services are special cased. They can be called, even though they do not satisfy the
-	// Callable interface.
-	if ref.APIVersion == "v1" && ref.Kind == "Service" {
-		url := &apis.URL{
-			Scheme: "http",
-			Host:   network.GetServiceHostname(ref.Name, ref.Namespace),
-			Path:   "",
-		}
-		return url, nil
-	}
-
 	addressable, ok := obj.(*duckv1.AddressableType)
 	if !ok {
 		return nil, apierrs.NewBadRequest(fmt.Sprintf("%+v (%T) is not an AddressableType", ref, ref))
 	}
+	if addressable.Status.Addresses != nil && len(addressable.Status.Addresses) != 0 {
+		if addressable.Status.Address == nil {
+			return &addressable.Status.Addresses[0], nil
+		} else {
+			for _, addr := range addressable.Status.Addresses {
+				if addr.Name == addressable.Status.Address.Name {
```

Comment:
In addition, we should change the condition of the `if` at line 255 to:
```
		if ref.Address == nil || *ref.Address == "" {
			return &addressable.Status.Addresses[0], nil
		}
```
so that we're in the case:
  - if dest.ref.address is not specified:
    - Select the first (in order) address from all addresses
    - If no address is found use status.address of the resolved destination